### PR TITLE
[ToDo] added edit button functionality along with jsdoc comments

### LINKED ISSUE #53: [To-Do] Due Dates
[To-Do] Due Dates
### Objectives
Have the todo tasks be able to support due dates
### Acceptance Criteria
- [ ] Can select new due date on create
- [ ] can change due date after created
- [ ] feel free to use warm up code
### Assignees
- @UKCSD 
- @shjucsd 
- @ChooChooChao 
- @alt012ucsd 
### Additional Notes 
please talk to each other about how you're gonna recycle code

### LINKED ISSUE #54: [To-Do] Modify + Add Tasks
[To-Do] Modify + Add Tasks
### Objectives
supporting adding new tasks and modify tasks
### Acceptance Criteria
- [ ] User is able to edit tasks 
- [ ] edit date
- [ ] edit description
- [ ] edit category
- [ ] edit title
- [x] User is able to create new tasks
- [ ] on creation, prompt for title and due date
- [ ] other fields are optional
### Assignees
- @UKCSD 
- @ChooChooChao 
- @shjucsd 
- @alt012ucsd

### PR DESCRIPTION
**Pull request by shjucsd**
# Objective
The objective of this PR is to integrate the functionality for edit button along with the existing todo features. Additionally, I have added jsdoc comments for all the functions as well.

### Solution
- Add JsDoc comments
- Updated the files: todo.html, todo.css, todo-app.js to accommodate the edit button functionality with pop-up modal
- This is how the pop-up after clicking on edit button looks like:
image

### Assignees
- @shjucsd 

### Additional Notes 
We still need to fix some minor issues regarding the display of edit and delete buttons when dragging and dropping to and from done column. Potentially, have a fallback option for drag and drop for mobile ui.

### CODE REVIEW

File: to_do/to-do-app.js [hunk 1]
```diff
@@ -344,9 +407,14 @@ function removeTask(id, category) {
...
 }
 
-
-function addTaskBetweenLists(id, fromCategory, toCategory){
-  
+/**
+ * Adds the task from the old column to new column after drop
+ *
+ * @param {number} id - The ID of the task.
+ * @param {number} oldCategory - The current category of the task.
```
Comment (kennethkietvuong) [inline to_do/to-do-app.js]: Change `oldCategory` to `fromCategory`

File: to_do/to-do-app.js [hunk 2]
```diff
@@ -344,9 +407,14 @@ function removeTask(id, category) {
...
+ * @param {number} newCategory - The new category of the task.
```
Comment (kennethkietvuong) [inline to_do/to-do-app.js]: Change `newCategory` to `toCategory`

**Review by kennethkietvuong — APPROVED** [review]
Great work on the edit button functionality and JSDoc function comments. There's one small nit-pick I have that I commented for a function, but besides that, it looks good!

**Comment by shjucsd** [comment]
corrected the jsdoc comment for addTaskBetweenLists function

**Review by adironene — APPROVED** [review]
amazing work with JSdocs

**Pull request merged**